api/v1beta3/auth_config_types.go

```diff
@@ -0,0 +1,897 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta3
+
+import (
+	k8score "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	// The following constants are used to identify the different methods of authentication.
+	UnknownAuthenticationMethod AuthenticationMethod = iota
+	ApiKeyAuthentication
+	JwtAuthentication
+	OAuth2TokenIntrospectionAuthentication
+	KubernetesTokenReviewAuthentication
+	X509ClientCertificateAuthentication
+	PlainIdentityAuthentication
+	AnonymousAccessAuthentication
+
+	// The following constants are used to identify the different methods of metadata fetching.
+	UnknownMetadataMethod MetadataMethod = iota
+	HttpMetadata
+	UserInfoMetadata
+	UmaResourceMetadata
+
+	// The following constants are used to identify the different methods of authorization.
+	UnknownAuthorizationMethod AuthorizationMethod = iota
+	PatternMatchingAuthorization
+	OpaAuthorization
+	KubernetesSubjectAccessReviewAuthorization
+	SpiceDBAuthorization
+
+	// The following constants are used to identify the different methods of auth response.
+	UnknownAuthResponseMethod AuthResponseMethod = iota
+	PlainAuthResponse
+	JsonAuthResponse
+	WristbandAuthResponse
+
+	// The following constants are used to identify the different methods of callback functions.
+	UnknownCallbackMethod CallbackMethod = iota
+	HttpCallback
+
+	// The following constants are used to identify the different types of credentials.
+	UnknownCredentialsType CredentialsType = iota
+	AuthorizationHeaderCredentials
+	CustomHeaderCredentials
+	QueryStringCredentials
+	CookieCredentials
+
+	// Status conditions
+	StatusConditionAvailable StatusConditionType = "Available"
+	StatusConditionReady     StatusConditionType = "Ready"
+
+	// Status reasons
+	StatusReasonReconciling     string = "Reconciling"
+	StatusReasonReconciled      string = "Reconciled"
+	StatusReasonInvalidResource string = "Invalid"
+	StatusReasonHostsLinked     string = "HostsLinked"
+	StatusReasonHostsNotLinked  string = "HostsNotLinked"
+	StatusReasonCachingError    string = "CachingError"
+	StatusReasonUnknown         string = "Unknown"
+
+	EvaluatorDefaultCacheTTL = 60
+)
+
+type AuthenticationMethod int8
+type MetadataMethod int8
+type AuthorizationMethod int8
+type AuthResponseMethod int8
+type CallbackMethod int8
+type CredentialsType int8
+
+type StatusConditionType string
+
+// AuthConfig is the schema for Authorino's AuthConfig API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
```

Comment:
Shall we remove it from `v1beta2`?

Reply:
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions